portfolio/src/main/webapp/script.js

```diff
@@ -216,3 +216,49 @@ function deleteAllComments() {
     }).then(getComments());
   }
 }
+
+function getUserInfo() {
+  fetch('/login')
+    .then(response => response.json())
+    .then((response) => {
+      const loginStatusElement = document.getElementById('login-status');
+      const commentForm = document.getElementById('comment-form');
+      const commentMessage = document.getElementById('comment-msg');
+
+      if (response.loggedIn) {
+        loginStatusElement.innerHTML =
+          `<div>Hello ${response.nickname}!</p>
+          <a href=${response.logoutUrl}>Log out</a>`;
+          commentForm.style.display = 'block';
+          commentMessage.innerHTML = '';
+      }
+      else {
+        loginStatusElement.innerHTML =
+          `<a href=${response.loginUrl}>Log in</a>`;
+        commentMessage.innerHTML = 'Click "Login" in the ' +
+            'top right to get the option to leave a comment.<br><br>';
+        
+      }
+    });
+}
+
+function setUserNickname() {
+  /* Check if user has already set a nickname */
+  fetch('/login')
+    .then(response => response.json())
+    .then((response) => {
+      if (response.nickname != "") {
+        window.location.replace("/");
+      }
+      /* If no nickname set, prompt for a nickname and save to datastore */
+      else {
+        let nickname = prompt('Enter a nickname to be displayed when you ' +
+          'leave a comment.');
+        /* TODO: don't allow user to cancel nickname input */
+        if(nickname == null) nickname = 'Anonymous';
```

Comment:
Here it looks like you assign "Anonymous" as the nickname of anyone who doesn't select one.  Maybe you could also do this in LoginServlet?